src/rustdoc_generation.rs

```diff
@@ -7,44 +7,50 @@ use crate::dump::RustDocCommand;
 use crate::util::slugify;
 use crate::GlobalConfig;
 
-pub(crate) trait BaselineLoader {
-    fn load_rustdoc(
+pub(crate) trait RustdocGenerator {
+    fn generate_rustdoc(
         &self,
         config: &mut GlobalConfig,
-        rustdoc: &RustDocCommand,
+        rustdoc_command: &RustDocCommand,
         name: &str,
-        version_current: Option<&semver::Version>,
+        highest_considered_registry_version: Option<&semver::Version>,
+        crate_type: &str,
```

Comment:
At this point, I feel like this trait has gotten complex enough that it needs documentation on what all its arguments mean. It's not obvious to me what `crate_type` is supposed to be, and `highest_considered_registry_version` has a specificity to it that makes me afraid that it might be more complex than it might seem at first glance.